### PR TITLE
Test that the generated Elm compiles

### PR DESCRIPTION
**Pull request by seagreen**
+ Create a temporary system directory with the `temporary` package
+ Generate Elm into it.
+ Place an "elm.json" file there so that the Elm will compile.
+ Test that it compiles with elm make

### CODE REVIEW

File: test/TestExampleServer.hs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+{-# language DeriveGeneric #-}
+{-# language FlexibleContexts #-}
+{-# language FlexibleInstances #-}
+{-# language MultiParamTypeClasses #-}
+{-# language OverloadedStrings #-}
+{-# language PolyKinds #-}
+{-# language ScopedTypeVariables #-}
+{-# language TypeApplications #-}
+{-# language TypeOperators #-}
+module TestExampleServer where
```
Comment (seagreen) [inline test/TestExampleServer.hs:13]: This module is roughly a copy of the example module (though with `main` removed`).

**Comment by seagreen** [comment]
I thought that before starting on https://github.com/folq/servant-to-elm/issues/9 it would be useful to have some tests, let me know what you think.

**Review by ollef — APPROVED** [review]
Nice, this seems like a good idea!

Note that there are more thorough tests for `servant-to-elm` in [haskell-to-elm-test](https://github.com/folq/haskell-to-elm-test).

File: servant-to-elm.cabal [hunk 2]
```diff
@@ -89,11 +90,20 @@ test-suite servant-to-elm-test
...
+    , directory >=1.3
     , elm-syntax >=0.3.0 && <0.3.1
+    , filepath >=1.4
+    , generics-sop >=0.4
     , haskell-to-elm >=0.3.0 && <0.3.2
+    , hspec >=2.7
     , http-types >=0.12.0
+    , prettyprinter >=1.2
+    , process >=1.6
     , servant >=0.16.0
```
[the remark this answers is not in the document]
Comment (seagreen) [reply servant-to-elm.cabal]: Sounds good, added the upper bound.

**Comment by seagreen** [comment]
> Nice, this seems like a good idea!
> 
> Note that there are more thorough tests for `servant-to-elm` in [haskell-to-elm-test](https://github.com/folq/haskell-to-elm-test).

Ah, that makes sense! I thought it was weird for this to be untested. Hopefully it will still be useful to have some local tests.

**Comment by ollef** [comment]
Hmm, that didn't seem to fix the issue. Do you have any ideas? https://travis-ci.com/github/folq/servant-to-elm/jobs/466242432

Otherwise we can probably merge this as is and fix CI/correct the version bounds later.

**Comment by seagreen** [comment]
Looks like the problem is due to this change in servant 0.17: https://github.com/haskell-servant/servant/blob/27173c922311112dd153346cf3cd72b9fb0f3551/servant/CHANGELOG.md#017

The fix will be adding a `HasElmEndpoints (NoContentVerb 'POST)` instance, but agreed it makes sense to do that separately from this PR.

**Comment by ollef** [comment]
Ah, but the weird thing is that the version bound doesn't help.

**Comment by seagreen** [comment]
> Ah, but the weird thing is that the version bound doesn't help.

Oh right, that is super weird.

**Comment by ollef** [comment]
I'm gonna go ahead with this, thanks!

**Pull request merged**